web/src/components/apps/AppVersionHistoryRow.jsx

```diff
@@ -119,10 +119,25 @@ function renderVersionStatus(version, app, match, viewLogs) {
   }
 
   let preflightsFailed = false;
-  if (version.status === "pending" && version.preflightResult) {
+  let preflightState = "";
+  if (version.preflightResult) {
     const preflightResult = JSON.parse(version.preflightResult);
-    const preflightState = getPreflightResultState(preflightResult);
-    preflightsFailed = preflightState === "fail";
+    if (version.status === "pending") {
+      preflightState = getPreflightResultState(preflightResult);
```

Comment:
this can be outside of the if statement since it's duplicated here https://github.com/replicatedhq/kots/pull/2055/files#diff-a5ee13382325550db7f0c09495c03715d7dbb7c581dfce6d03a32d0ee81360cdR129

Reply:
ah yeah good catch. Updated this logic a bit and didn't move that part out.